clients/drcachesim/tracer/output.cpp

```diff
@@ -1340,11 +1347,19 @@ exit_thread_io(void *drcontext)
             BUF_PTR(data->seg_base), dr_get_thread_id(drcontext));
 
         ptr_int_t window = get_local_window(data);
-        process_and_output_buffer(drcontext,
-                                  /* If this thread already wrote some data, include
-                                   * its exit even if we're over a size limit.
-                                   */
-                                  data->bytes_written > 0);
+        if (has_tracing_windows()) {
+            if (is_in_tracing_mode(tracing_mode.load(std::memory_order_acquire)) &&
+                get_local_window(data) == window) {
```

Comment:
Isn't this always true?  We just set window on line 1349.

Reply:
Right. I have removed it.